webapi/Controllers/ChatHistoryController.cs

```diff
@@ -122,22 +129,25 @@ public async Task<IActionResult> GetChatSessionByIdAsync(Guid chatId)
     }
 
     /// <summary>
-    /// Get all chat sessions associated with a user. Return an empty list if no chats are found.
-    /// The regex pattern that is used to match the user id will match the following format:
-    ///    - 2 period separated groups of one or more hyphen-delimited alphanumeric strings.
-    /// The pattern matches two GUIDs in canonical textual representation separated by a period.
+    /// Get all chat sessions associated with the logged in user. Return an empty list if no chats are found.
     /// </summary>
-    /// <param name="userId">The user id.</param>
     /// <returns>A list of chat sessions. An empty list if the user is not in any chat session.</returns>
     [HttpGet]
     [Route("chatSession/getAllChats/{userId:regex(([[a-z0-9]]+-)+[[a-z0-9]]+\\.([[a-z0-9]]+-)+[[a-z0-9]]+)}")]
     [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
     public async Task<IActionResult> GetAllChatSessionsAsync(string userId)
     {
+        // TODO: [Issue #141] Remove this once we remove userId from route
+        if (!userId.Equals(this._authInfo.UserId, StringComparison.Ordinal))
+        {
+            return this.Forbid("User id does not match request.");
+        }
+
```

Comment:
I think this can be removed. We use the userId that comes from the auth header value anyway so the userId in the route is just a temporary placeholder to keep things backward compatible. We don't really care what the value is.